### PR TITLE
Bugfix/130 sugrinov fix failing integration tests

### PR DESCRIPTION
**Pull request by sugrinov**
* Changed authentication config
* Fixed tests that were failing

### CODE REVIEW

File: core/src/main/java/eu/execom/testgenerator/core/model/enumeration/UserRole.java [hunk 1]
```diff
@@ -1,11 +1,16 @@
 package eu.execom.testgenerator.core.model.enumeration;
 
 public enum UserRole {
-  ADMINISTRATOR,
-  REVIEWER,
-  APPLICANT;
+  ROLE_ADMINISTRATOR,
+  // TODO sugrinov: 30.5.18. Check if we need reviewer role at all.
```
Comment (stanislavgrujic) [inline core/src/main/java/eu/execom/testgenerator/core/model/enumeration/UserRole.java]: Yes, it is needed. Please remove comment.

File: web/src/main/java/eu/execom/testgenerator/web/config/OAuth2Config.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+     * @param map logged in user info.
+     * @return roles based on user info.
+     */
+    @Override
+    public List<GrantedAuthority> extractAuthorities(Map<String, Object> map) {
+
+      String domain = (String) map.get(HOST_DOMAIN_KEY);
+
+      final List<String> authorities = new ArrayList<>();
+      authorities.add("ROLE_USER");
```
Comment (stanislavgrujic) [inline web/src/main/java/eu/execom/testgenerator/web/config/OAuth2Config.java]: Please use instead
```
List<String> authorities = Arrays.asList(UserRole.USER.name());
```

File: web/src/main/java/eu/execom/testgenerator/web/config/OAuth2Config.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+     * Override {@link FixedAuthoritiesExtractor} to support custom roles.
+     * @param map logged in user info.
+     * @return roles based on user info.
+     */
+    @Override
+    public List<GrantedAuthority> extractAuthorities(Map<String, Object> map) {
+
+      String domain = (String) map.get(HOST_DOMAIN_KEY);
+
+      final List<String> authorities = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (sugrinov) [reply web/src/main/java/eu/execom/testgenerator/web/config/OAuth2Config.java]: It's not copied. Intellij automatically adds final when extracting variables. I will remove it

File: core/src/main/java/eu/execom/testgenerator/core/model/enumeration/UserRole.java [hunk 4]
```diff
@@ -1,11 +1,16 @@
...
+  ROLE_REVIEWER,
+  ROLE_APPLICANT;
```
[the remark this answers is not in the document]
Comment (sugrinov) [reply core/src/main/java/eu/execom/testgenerator/core/model/enumeration/UserRole.java]: OAuth2 checks to see if `ROLE_` exists in the String. In order to support it's authentication we need to have roles declared like this.

File: web/src/main/java/eu/execom/testgenerator/web/config/OAuth2Config.java (hunk 2 above)
Comment (sugrinov) [reply web/src/main/java/eu/execom/testgenerator/web/config/OAuth2Config.java]: Isn't `Arrays.asList` immutable?

Comment (stanislavgrujic) [reply web/src/main/java/eu/execom/testgenerator/web/config/OAuth2Config.java]: Feel free to check the code, it's open source... :)

Comment (sugrinov) [reply web/src/main/java/eu/execom/testgenerator/web/config/OAuth2Config.java]: This is from AbstractList which ArrayList inside Arrays class extends
```
public void add(int var1, E var2) {
  throw new UnsupportedOperationException();
}
```
Arrays.asList
```
public static <T> List<T> asList(T... var0) {
    return new Arrays.ArrayList(var0);
}
```

**Comment by stanislavgrujic** [comment]
Please resolve merge conflicts.

**Comment by sugrinov** [comment]
This may not be implemented yet

**Pull request closed without merging**